files/__addonLocation__/rollup.config.mjs

```diff
@@ -67,5 +71,37 @@ export default {
 <% filesToCopyFromRootToAddon.forEach((file) => { %>        { src: '<%= pathFromAddonToRoot %>/<%= file %>', dest: '.' },
 <% }); %>      ],
     }),
-<% } %>  ],
+<% } %>
+
+<% if (typescript) { %>
+    {
+      name: "Build Declarations",
+      closeBundle: async () => {
+        /**
+         * Generate the types (these include /// <reference types="ember-source/types"
+         * but our consumers may not be using those, or have a new enough ember-source that provides them.
+         */
+        console.log("Building types");
+        <% if (npm) { %>
+        await $({ stdio: 'inherit' })`npm exec glint -- --declaration`;
+        <% } else { %>
+        await $({ stdio: 'inherit' })`<%= packageManager %> glint --declaration`;
+        <% } %>
+
+        /**
+         * https://github.com/microsoft/TypeScript/issues/56571#
+         * README: https://github.com/NullVoxPopuli/fix-bad-declaration-output
+         */
+        console.log("Fixing types");
+        await fixBadDeclarationOutput("declarations/**/*.d.ts", [
+          // https://github.com/microsoft/TypeScript/issues/56571#issuecomment-1830436576
+          ["TypeScript#56571", { types: "all" }],
+          // https://github.com/typed-ember/glint/issues/628
+          "Glint#628",
+        ]);
+        console.log("⚠️ Dangerously (but neededly) fixed bad declaration output from typescript");
+      },
+    },
+<% } %>
+  ],
```

Comment:
I feel like this plugin shouldn't be exploded inline into addon's rollup configs 🤔 should it be a new dependency that just provides this? or is it just something that the Embroider addon-dev is providing as a plugin (like the other ones in this file)?

Reply:
addon-dev _could_ provide it, if it weren't using `tsc` to compile (because the way embroider is configured to use `tsc` right now doesn't actually support `await import()` in CJS).

I would just call it `addon.declarations(/* default output "declarations" folder */ )` or something like that.

Reply:
I'd also like to see the Rollup config remain simple and dependencies to a minimum. If I understood the release notes right, the reference directives won't appear with `typescript@5.5`?

- https://github.com/microsoft/TypeScript/pull/57681
- https://devblogs.microsoft.com/typescript/announcing-typescript-5-5-rc/#simplified-reference-directive-declaration-emit

If so, the blueprint could remain as is, and only those who are on `typescript@<=5.4` could install `fix-bad-declaration-output` as needed.

Reply:
Naw glint is still borked out of the box